app/Services/ApiService.php

```diff
@@ -0,0 +1,45 @@
+<?php
+
+namespace App\Services;
+
+use Illuminate\Support\Facades\Http;
+use Mockery\Exception;
+
+class ApiService
+{
+    public string $url;
+    public array $params = [];
+    public array $baseHeaders = [
+        'Access-Control-Request-Method' => 'application/json',
+        'content-type' => 'application/json',
+    ];
+    public function __construct()
+    {
+        $this->url = config('base.base_url');
+    }
+
+    public function addParam($param){
+        $this->params = array_merge($this->params, $param);
+        return $this;
+    }
+
+    public function addHeader($key, $value)
+    {
+        $this->baseHeaders[$key] = $value;
+        return $this;
+    }
+
+    function sendRequest($endPoint){
+        $baseFilter = [
+            'site' => 'stackoverflow',
+        ];
+
+        $baseFilter = array_merge($baseFilter, $this->params);
+
+        $detailUrl = $this->url.config('base.base_version').$endPoint;
```

Comment:
აქ ისე ქენი რო ორივე ერთნაირად იყოს $this->url და config( დან წაგითხვა, 
ალბათ base_version -იც url -ის მსგავსაც ჯობია გაიტანო